src/nbsap/forms.py

```diff
@@ -47,11 +47,10 @@ def to_python(self, value):
 class ChoicesMixin(object):
     def _get_choices(self, name, queryset, attr):
         return sorted(
-            [(x.pk, '{} {}'.format(name, getattr(x, attr)))
+            [(x.pk, name + ' ' + '-'.join([str(getattr(x, a)) for a in attr]))
              for x in queryset],
             key=lambda el: el[1].split()[1])
 
```

Comment:
Quite risky changing the `attr` parameter type and then replacing it in every `_get_choices` call. Why don't check if attr is a string inside the function, and then, locally, convert it to a list with one element?